tapdb/assets_store.go

```diff
@@ -2445,22 +2445,44 @@ func (a *AssetStore) ConfirmParcelDelivery(ctx context.Context,
 		for idx := range outputs {
 			out := outputs[idx]
 
-			isNumsKey := bytes.Equal(
-				out.ScriptKeyBytes, asset.NUMSBytes,
+			// Decode the witness first, so we can find out if this
+			// is a burn or not.
+			var witnessData []asset.Witness
+			err = asset.WitnessDecoder(
+				bytes.NewReader(out.SerializedWitnesses),
+				&witnessData, &[8]byte{},
+				uint64(len(out.SerializedWitnesses)),
+			)
+			if err != nil {
+				return fmt.Errorf("unable to decode "+
+					"witness: %w", err)
+			}
+
+			scriptPubKey, err := btcec.ParsePubKey(
+				out.ScriptKeyBytes,
 			)
+			if err != nil {
+				return fmt.Errorf("unable to decode script "+
+					"key: %w", err)
+			}
+
+			isNumsKey := scriptPubKey.IsEqual(asset.NUMSPubKey)
 			isTombstone := isNumsKey &&
 				out.Amount == 0 &&
 				out.OutputType != int16(
 					tappsbt.TypePassiveAssetsOnly,
 				)
+			isBurn := !isNumsKey && len(witnessData) > 0 &&
```

Comment:
Confused here: thought burning was _just_ sending to the output, with no special witness needed?

Reply:
Nvm needed for if the asset was a split.